### DISCUSSION TITLE
[Flutter] How to test deep links?

### DISCUSSION THREAD

**Discussion by shrijanRegmi** [mobile-dev-inc/Maestro#1740] [Q&A]
Suppose, I have a deep link that someone shared on a social media platform. The link should open the app and take the user to a specific screen. How do I test this flow?

**Comment by tiagodread** [comment]
Example:

```
---
appId: sample
name: Sample flow

---
- openLink: apppackage://youur/internal/deep/link

- assertVisible: "welcome to sample"

- assertVisible: "Foo"
```

**Reply by ushavaddi** [reply]
This solution works for internal deep linking. How about for external
Deep linking testing using iOS devices?  External deep linking starts with https://

**Comment by krish-techx** [comment]
In addition to internal deep links, here’s a complete way to test **both internal and external (https) deep links**, including iOS.

---

### 1. Internal Deep Links (Custom Schemes)

These are links like:

```
apppackage://your/internal/path
```

You can test them directly in Maestro:

```yaml
- openLink: apppackage://your/internal/deep/link
```

---

### 2. External Deep Links (https / Universal Links)

These represent real-world scenarios (e.g., links opened from Safari, WhatsApp, etc.).

#### Testing methods:

**a) Safari (real device)**

* Paste your link (e.g., `https://your-domain.com/path`) into Safari
* If Universal Links are configured correctly, it will open your app

**b) iOS Simulator**

```bash
xcrun simctl openurl booted https://your-domain.com/path
```

**c) Maestro**

```yaml
- openLink: https://your-domain.com/path
```

---

### 3. Required Setup for iOS (Universal Links)

Make sure:

* Associated Domains are enabled in Xcode (`applinks:your-domain.com`)
* `apple-app-site-association` file is correctly hosted
* The app is reinstalled after changes (iOS caches link associations)

---

### 4. Common Pitfalls

* If the link opens in Safari instead of the app → Universal Links are not configured correctly
* Domain is not properly verified
* Changes not reflected due to caching (reinstall the app)

---

### Summary

* Internal → `app://` (custom schemes)
* External → `https://` (Universal Links)
* Test using Maestro, Safari, or simulator depending on your workflow
